net/udprelay/server.go

```diff
@@ -0,0 +1,517 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package udprelay
+
+import (
+	"bytes"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"net"
+	"net/netip"
+	"strconv"
+	"sync"
+	"time"
+
+	"go4.org/mem"
+	"tailscale.com/disco"
+	"tailscale.com/net/packet"
+	"tailscale.com/types/key"
+)
+
+const (
+	defaultBindLifetime        = time.Second * 5
```

Comment:
is this how long before a binding times out?

Reply:
if a client was on a plane, we've seen RTT's pushing 1.5s, we might need a bit more to complete the whole handshake

Reply:
Yea, this was somewhat arbitrary and probably too low. Updated with justification in https://github.com/tailscale/tailscale/pull/15480/commits/9860f1e31610624eeddee2b8b708b97911a40d02.